test/e2e/aws_test.go

```diff
@@ -185,4 +198,69 @@ func TestAWS(t *testing.T) {
 		}
 	}
 
+	// Check that the aws access key and secret key in the system configuration
+	// can be overridden by a user defined secret
+	fakeAWSCredsData := map[string]string{
+		"REGISTRY_STORAGE_S3_ACCESSKEY": "myAccessKey",
+		"REGISTRY_STORAGE_S3_SECRETKEY": "mySecretKey",
+	}
+
+	if _, err := util.CreateOrUpdateSecret(regopapi.ImageRegistryPrivateConfigurationUser, testframework.ImageRegistryDeploymentNamespace, fakeAWSCredsData); err != nil {
+		t.Fatalf("unable to create secret %q: %#v", fmt.Sprintf("%s/%s", testframework.ImageRegistryDeploymentNamespace, regopapi.ImageRegistryPrivateConfigurationUser), err)
+	}
+
+	cfgUser, err := clusterconfig.GetAWSConfig()
+	if err != nil {
+		t.Errorf("unable to get aws configuration: %#v", err)
+	}
+	if fakeAWSCredsData["REGISTRY_STORAGE_S3_ACCESSKEY"] != cfgUser.Storage.S3.AccessKey || fakeAWSCredsData["REGISTRY_STORAGE_S3_SECRETKEY"] != cfgUser.Storage.S3.SecretKey {
+		t.Errorf("expected system configuration to be overridden by the user configuration but it wasn't.")
+	}
+
+	// Ensure that a useful status condition is set on the image registry resource
+	// if we can't create the s3 bucket because of invalid aws credentials
+	testframework.MustRemoveImageRegistry(t, client)
+	testframework.DeployImageRegistry(t, client, nil)
+
+	err = wait.Poll(1*time.Second, testframework.AsyncOperationTimeout, func() (stop bool, err error) {
+		// Get a fresh version of the image registry resource
+		cr, err = client.ImageRegistries().Get(testframework.ImageRegistryName, metav1.GetOptions{})
+		if err == nil {
+			return true, nil
+		}
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+
+		return true, err
+	})
+	if err != nil {
+		testframework.DumpImageRegistryResource(t, client)
+		testframework.DumpOperatorLogs(t, client)
+		t.Fatal(err)
+	}
+
+	foundStorageExists := false
+	for _, condition := range cr.Status.Conditions {
+		if condition.Type == regopapi.StorageExists {
+			foundStorageExists = true
+			if condition.Status != operatorapi.ConditionFalse {
+				t.Errorf("condition StorageExists should be \"false\" but was %v instead.", condition.Status)
+
+			}
+			if condition.Reason != "InvalidAccessKeyId" {
+				t.Errorf("condition Reason should have been \"InvalidAccessKeyId\" but was %s instead.", condition.Reason)
+			}
+		}
+	}
+	if !foundStorageExists {
+		t.Errorf("condition StorageExists was not found, but should have been. %#v", cr.Status.Conditions)
+	}
+
+	// Clean up the image-registry-private-configuration-user secret
+	err = client.Secrets(testframework.ImageRegistryDeploymentNamespace).Delete(regopapi.ImageRegistryPrivateConfigurationUser, &metav1.DeleteOptions{})
+	if err != nil {
+		t.Errorf("unable to remove %s/%s: %#v", testframework.ImageRegistryDeploymentNamespace, regopapi.ImageRegistryPrivateConfigurationUser, err)
+	}
+
```

Comment:
need a test that confirms when we delete the registry resource, the finalizer cleans up all resources + the s3 buckets.